File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s=list(s)
        vowels=['a','e','i','o','u','A','E','I','O','U']
        vowel_array=[]
        for i in range(len(s)):
            if s[i] in vowels:
                vowel_array.append(s[i])
        vowel_array=vowel_array[::-1]
        j=0
        for i in range(len(s)):
            if s[i] in vowels:
                s[i]=vowel_array[j]
                j+=1
        return "".join(s)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isvowel(char ch){
        ch=tolower(ch);
        return ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u';
    }
    string reverseVowels(string s) {
        int l=0;
        int r=s.size()-1;
        while(l<r){
            if(isvowel(s[l]) && isvowel(s[r])){
                swap(s[l],s[r]);
                l++;
                r--;
            }
            else if(isvowel(s[l])==0){
                l++;
            }
            else{
                r--;
            }
        }
        return s;
    }
};